runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -502,7 +502,17 @@ static void setupAvailableParallelism(int32_t maxThreads) {
     if (hwpar > 0) {
         // Limit the parallelism to the maximum imposed by the comm layer.
         if (0 < maxThreads && maxThreads < hwpar) {
-            hwpar = maxThreads;
+          if (chpl_nodeID == 0) {
+            char msg[1024];
+            snprintf(msg, sizeof(msg),
+                     "The CHPL_COMM setting is limiting the number of threads "
+                     "to %d, rather than the hardware's preference of %d.  "
+                     "If using CHPL_COMM=gasnet, set 'GASNET_MAX_THREADS' to "
+                     "a higher number in your environment to address this.\n",
+                     maxThreads, hwpar);
+            chpl_warning(msg, 0, 0);
```

Comment:
New warning text LGTM, and should usually achieve the goal of avoiding silent under-utilization of cores.

However IIUC the new warning won't be printed on a heterogeneous system where node 0 has a smaller core count than `maxThreads`, but some other node might have a higher core count.

Not sure if such node heterogeneity is of interest (or even permitted by Chapel?) so feel free to ignore this observation if it seems irrelevant or not worth addressing.